File: src/main/java/com/mexchange/api/repo/UsersRepository.java
package com.mexchange.api.repo;

import com.mexchange.api.entity.User;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.rest.core.annotation.RepositoryRestResource;

@RepositoryRestResource(collectionResourceRel = "users", path = "users")
public interface UsersRepository extends MongoRepository<User, String> {
}

File: src/main/kotlin/com/mexchange/api/repo/UsersRepository.kt
package com.mexchange.api.repo

import com.mexchange.api.entity.User
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.data.rest.core.annotation.RepositoryRestResource

@RepositoryRestResource(collectionResourceRel = "users", path = "users")
interface UsersRepository : MongoRepository<User, String>
